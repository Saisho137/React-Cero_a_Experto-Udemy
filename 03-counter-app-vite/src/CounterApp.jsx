import { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const handleSumClick = () => {
        //console.log(event)
        setCounter(counter + 1);
    };

    const handleSubstractClick = () => {
        //console.log(event)
        setCounter(counter - 1);
    };

    const handleResetClick = () => {
        //console.log(event)
        setCounter(value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>

            <button onClick={handleSumClick}> +1 </button>
            <button onClick={handleSubstractClick}> -1 </button>
            <button onClick={handleResetClick}> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

export default CounterApp;