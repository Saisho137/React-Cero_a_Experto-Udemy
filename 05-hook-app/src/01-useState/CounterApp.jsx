import { useState } from "react"

const CounterApp = () => {
    const [state, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
    });

    const { counter1, counter2, counter3 } = state;

    return (
        <>
            <h1>Counter 1: {counter1}</h1>
            <h2>Counter 2: {counter2}</h2>
            <h3>Counter 3: {counter3}</h3>
            <hr />
            <button className="btn" onClick={() => setCounter({
                ...state,
                counter1: counter1 + 1
            })}>+1</button>
        </>
    )
}

export default CounterApp
