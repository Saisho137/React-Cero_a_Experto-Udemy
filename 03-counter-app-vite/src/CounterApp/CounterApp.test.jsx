import { render, screen, fireEvent } from '@testing-library/react';
import CounterApp from "./CounterApp";

describe('Pruebas para CounterApp', () => {
    const value = 100;

    test('Deberia hacer el match con el snapshot', () => {
        const { containter } = render(<CounterApp value={value} />);
        expect(containter).toMatchSnapshot();
    });

    test('Deberia mostrar el valor inicial de 100', () => {
        const { containter } = render(<CounterApp value={value} />);
        expect(screen.getByText(value).innerHTML).toContain("100");
    });

    test('Deberia aumtentar el contado al hacer click en "+1"', () => {
        render(<CounterApp value={value} />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();
    });

    test('Deberia decrementar el contado al hacer click en "-1"', () => {
        render(<CounterApp value={value} />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();
    });

    test('Deberia funcionar el boton de "reset"', () => {
        render(<CounterApp value={value} />);

        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('-1'));

        expect(screen.getByText('97')).toBeTruthy();

        fireEvent.click(screen.getByRole('button', { name: 'Reset' }));
        expect(screen.getByText(value)).toBeTruthy();
    });
})
