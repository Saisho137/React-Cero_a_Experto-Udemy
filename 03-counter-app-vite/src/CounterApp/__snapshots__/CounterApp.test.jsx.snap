// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para CounterApp Deberia hacer el match con el snapshot 1`] = `undefined`;
